samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs

```diff
@@ -160,47 +162,56 @@ public async Task ReassignWorkflow()
-                    batchRequestContent.AddBatchRequestStep(
-                            new BatchRequestStep(
-                                // Use the current assignment as id for this step
-                                assignment.Id,
-                                // The step takes the HttpRequestMessage from the request
-                                graphClient.Education
+                    // Use the request builder to generate a regular request to get the assignment submissions
+                    var asgSubmissionsRequest = graphClient.Education
                                     .Classes[assignment.ClassId]
                                     .Assignments[assignment.Id]
                                     .Submissions
-                                    .Request()
-                                    .GetHttpRequestMessage())
-                        );
+                                    .ToGetRequestInformation();
+
+                    // Create HttpRequestMessage for the regular request
+                    var eventsRequestMessage = await graphClient.RequestAdapter.ConvertToNativeRequestAsync<HttpRequestMessage>(
+                        asgSubmissionsRequest
+                     );
+
+                    // Adds each request to the batch
+                    batchRequestContent.AddBatchRequestStep(
+                        new BatchRequestStep(
+                            // Use the current assignment as id for this step
+                            assignment.Id,
+                            // The step takes the HttpRequestMessage from the request
+                            eventsRequestMessage)
+                    );
                 }
 
                 // Build a return response object for our batch
-                var returnedResponse = await graphClient.Batch.Request().PostAsync(batchRequestContent);
-                
-                foreach (var assignment in meAssignments)
+                var returnedResponse = await graphClient.Batch.PostAsync(batchRequestContent);
+
+                foreach (var assignment in meAssignments.Value)
                 {
                     Console.WriteLine($"Getting assignment {assignment.Id} submissions");
 
                     // De-serialize the response based on return type
-                    var submissionsResponse = await returnedResponse.GetResponseByIdAsync<EducationAssignmentSubmissionsCollectionResponse>(assignment.Id);
+                    var submissionsResponse = await returnedResponse.GetResponseByIdAsync<EducationSubmissionCollectionResponse>(assignment.Id);
 
                     // Get and print submissions (if any)
                     if (submissionsResponse == null) continue;
 
                     // "Value" contains the request response
                     var submissions = submissionsResponse.Value;
-                    foreach (var submission in submissions)
+                    foreach (var submission in submissions!)
```

Comment:
if we're not going to handle null, I'd suggest removing the nullable/enable in the .csproj files

Reply:
got it, removed from project files